### PR TITLE
Fix bug when being muted by a moderator while on push to talk

### PR DESCRIPTION
**Pull request by ma12-co**
Fix: #4368 

Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
Code makes sense, I've tested the UI states and they looked fine :+1:

File: src/components/CallView/shared/LocalMediaControls.vue [hunk 1]
```diff
@@ -186,6 +186,8 @@ export default {
 			boundaryElement: document.querySelector('.main-view'),
 			isQualityWarningTooltipDismissed: false,
 			mouseover: false,
+			audioEnabledBeforeSpacebarKeydown: undefined,
+			spacebarKeyDown: false,
```
Comment (danxuliu) [inline src/components/CallView/shared/LocalMediaControls.vue:190]: (Independently of the main review) I would rename `spacebarKeyDown` to something more generic like `isPushToTalkActive`.

**Review by danxuliu — DISMISSED** [review]
I would generalize the current approach so stopping push to talk/mute only restores the previous state if the audio state has not changed in the meantime.

Scenario 1:

- Enable audio
- Hold spacebar to mute
- A moderator force mutes all the users
- Release spacebar

Scenario 2:

- Disable audio
- Hold spacebar to unmute
- Click the mute button
- Release spacebar

In both scenarios (quite corner cases, yes :-) ) the participant will be unmuted. In my opinion it would be better to not restore the previous unmuted state when the spacebar is released.

In order to do that (and also handle the opposite of scenario 3, enable audio, hold to mute, click unmute, release -> muted again) I suggest something like (untested):
```
methods: {
    handleShortKey() {
        this.isPushToTalkActive = !this.isPushToTalkActive

        if (this.isPushToTalkCanceled) {
            this.isPushToTalkCanceled = false
            return
        }

        this.toggleAudio()
}

watch: {
    'this.model.attributes.audioEnabled': function() {
        if (this.isPushToTalkActive) {
            this.isPushToTalkCanceled = true
        }
    }
}
```

Changes will be needed too to handle `forceMute` events, but I will spare you the details until you have confirmed that what I suggest is the way to go :-P

There is an additional issue (no matter the approach used), but I think it could be ignored (unless there is an easy way to handle it): if you hold the spacebar to unmute and then click on the chat input the input gets the key down events and releasing the spacebar does not cause `handleShortKey` to be called again.

**Comment by ma12-co** [comment]
Shouldn't we just disable the mute/unmute button when hitting the spacebar?

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
So we tested it a lot and the watching didn't really work too good.
We now made it so your state after releasing the spacebar is what the user expects and that is (at least from my pov) what it should do.

File: src/components/CallView/shared/LocalMediaControls.vue [hunk 2]
```diff
@@ -170,6 +170,8 @@ export default {
 			mouseover: false,
 			callAnalyzer: callAnalyzer,
 			qualityWarningInGracePeriodTimeout: null,
+			audioEnabledBeforeSpacebarKeydown: undefined,
+			spacebarKeyDown: false,
```
Comment (danxuliu) [inline src/components/CallView/shared/LocalMediaControls.vue:174]: `isPushToTalkActive` became `spacebarKeyDown` again :-(

Comment (ma12-co) [reply src/components/CallView/shared/LocalMediaControls.vue:174]: yes, because it can also be push to mute. That's why we decided to change it back.

**Pull request merged**